lib/workers/repository/init/merge.ts

```diff
@@ -264,3 +277,24 @@ export async function mergeRenovateConfig(
   }
   return returnConfig;
 }
+
+export function logShallowConfig(ShallowConfig: RenovateConfig): void {
+  // remove resolved external presets from the extends array for the log
+  // clean duplicate presets in case of two different presets are having
+  // same values in the extends array
+  if (is.array(ShallowConfig?.extends)) {
+    if (ShallowConfig.extends.length) {
+      ShallowConfig.extends = ShallowConfig.extends.filter(
+        (e) => !isPresetForShallowConfig(e)
+      );
+      const uniqueExtends = new Set(ShallowConfig.extends.values());
+      ShallowConfig.extends = Array.from(uniqueExtends);
+    }
+    // istanbul ignore if
+    if (ShallowConfig.extends.length === 0) {
+      delete ShallowConfig.extends;
+    }
+  }
```

Comment:
Why can't this filtering logic be part of `resolveConfigPresets()`?

Ideally in this file we just do:
- resolve the config shallowly
- log it